tracer/src/Datadog.Trace.SourceGenerators/Configuration/ConfigurationKeysGenerator.cs

```diff
@@ -0,0 +1,767 @@
+            var seeAlsoLines = new List<string>();
+            foreach (var line in docLines)
+            {
+                var trimmedLine = line.TrimStart();
+
+                // Check if line contains seealso tag (self-closing /> or closing </seealso>) as we need to extract it
+                if (trimmedLine.StartsWith("<seealso") && (trimmedLine.Contains("/>") || trimmedLine.Contains("</seealso>")))
+                {
+                    // seealso tags go outside summary - save for later
+                    seeAlsoLines.Add(line.Trim());
+                    continue;
+                }
+
+                sb.AppendLine($"{indent}/// {line}");
+            }
+
+            sb.AppendLine($"{indent}/// </summary>");
+
+            // Add seealso tags after summary
+            foreach (var seeAlsoLine in seeAlsoLines)
+            {
+                sb.AppendLine($"{indent}/// {seeAlsoLine}");
+            }
+        }
+
+        // Add Obsolete attribute if deprecated
+        if (!string.IsNullOrEmpty(entry.DeprecationMessage))
+        {
+            // Escape quotes in the deprecation message and trim whitespace
+            var escapedMessage = entry.DeprecationMessage!.Trim().Replace("\"", "\\\"");
+            sb.AppendLine($"{indent}[System.Obsolete(\"{escapedMessage}\")]");
+        }
+
+        // Add const declaration
+        var constName = KeyToConstName(entry.Key, ProductNameEquivalents(productName), nameMapping);
+        sb.AppendLine($"{indent}public const string {constName} = \"{entry.Key}\";");
+    }
+
+    private static string KeyToConstName(string key, string[]? productNames = null, Dictionary<string, ConstantNameMapping>? nameMapping = null)
+    {
+        // First, check if we have a mapping for this key
+        if (nameMapping != null && nameMapping.TryGetValue(key, out var mapping))
+        {
+            // Use the mapped name from ConfigurationKeys
+            return mapping.ConstantName;
+        }
+
+        // Fallback to the original implementation
+        // Remove DD_ or OTEL_ prefix
+        var name = key;
+        var prefixes = new[] { "DD_", "_DD_" };
+        foreach (var prefix in prefixes)
+        {
+            if (name.StartsWith(prefix) && !char.IsDigit(name[0]))
+            {
+                var result = name.Substring(prefix.Length);
+                if (!char.IsDigit(result[0]))
+                {
+                    name = result;
+                }
+
+                break;
+            }
+        }
+
+        // Convert to PascalCase
+        var parts = name.Split('_');
+        var pascalName = string.Concat(parts.Select(p => p.Length > 0 ? char.ToUpper(p[0]) + p.Substring(1).ToLower() : string.Empty));
+
+        // Strip product prefix if the const name starts with it
+        if (productNames != null)
+        {
+            foreach (var productName in productNames)
+            {
+                if (pascalName!.Length > productName.Length &&
```

Comment:
`pascalName` can't be null?

```suggestion
                if (pascalName.Length > productName.Length &&
```